docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,382 @@
+Signature:
 
-There is a fundamental difference between the C#/VB async model and the F# async model.
+```fsharp
+computations: seq<Async<'T>> -> Async<'T[]>
+```
 
-When you call a function which returns a `Task` or `Task<'T>`, that job has already begun execution. The handle returned represents an already-running asynchronous job. In contrast, when you call an async function in F#, the `Async<'a>` returned represents a job which will be **generated** at some point. Understanding this model is powerful, because it allows for asynchronous jobs in F# to be chained together easier, performed conditionally, and be started with a finer grain of control.
+When to use it:
 
-There are a few other similarities and differences worth noting.
+- If you need to execute multiple computations in order and/or you rely on their order of execution.
 
-### Similarities
+What to watch out for:
 
-* `let!`, `use!`, and `do!` are analogous to `await` when calling an async job from within an `async{ }` block.
+- Computations will be run in the order that they are passed to this function, which can mean significantly more time spent to get a result when compared with `Async.Parallel`.
 
-  The three keywords can only be used within an `async { }` block, similar to how `await` can only be invoked inside an `async` method. In short, `let!` is for when you want to capture and use a result, `use!` is the same but for something whose resources should get cleaned after it’s used, and `do!` is for when you want to wait for an async workflow with no return value to finish before moving on.
+### Async.AwaitTask
 
-* F# supports data-parallelism in a similar way.
+Returns an asynchronous computation that waits for the given <xref:System.Threading.Tasks.Task%601> to complete and returns its result as an `Async<'T>`
 
-  Although it operates very differently, `Async.Parallel` corresponds to `Task.WhenAll` for the scenario of wanting the results of a set of async jobs when they all complete.
+Signature:
 
-### Differences
+```fsharp
+task: Task<'T>  -> Async<'T>
+```
 
-* Nested `let!` is not allowed, unlike nested `await`
+When to use:
 
-  Unlike `await`, which can be nested indefinitely, `let!` cannot and must have its result bound before using it inside of another `let!`, `do!`, or `use!`.
+- When you are consuming a .NET API that returns a <xref:System.Threading.Tasks.Task%601> within an F# asynchronous computation.
 
-* Cancellation support is simpler in F# than in C#/VB.
+What to watch out for:
 
-  Supporting cancellation of a task midway through its execution in C#/VB requires checking the `IsCancellationRequested` property or calling `ThrowIfCancellationRequested()` on a `CancellationToken` object that’s passed into the async method.
+- This call will allocate an additional `Async<'T>` object, which can increase overhead if it is used often.
```

Comment:
this is probably insignificant in the context of async and computation expressions in general. I would probably remove.